internal/version/version_test.go

```diff
@@ -0,0 +1,18 @@
+package version_test
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/artefactual-sdps/preprocessing-moma/internal/version"
+)
+
+func TestVersion(t *testing.T) {
+	// The results below lack vcs build info due to bug:
+	// https://github.com/golang/go/issues/33976, and the output should include
+	// the expected git data when the bug is fixed.
+	assert.Equal(t, version.Short, "0.0.0-dev")
+	assert.Equal(t, version.Long, "0.0.0-dev-t")
+	assert.Equal(t, version.Info("testapp"), "testapp version 0.0.0-dev-t")
```

Comment:
Can we fake the VERSION.txt file so we don't have to update this test each time the version changes?

Reply:
> Can we fake the VERSION.txt file so we don't have to update this test each time the version changes?

I don't know but it sounds tricky, since embedding, vars and init() all happens before the test runs.
You could try to fork a Go child process to test again after you change the contexts of VERSION.txt but it's not worth the effort. This is to me one of the rare cases where it's justifiable to not have a test. For some context context, it's inspired in an old version of the version package in the tailscaled project where they similarly opted not to include tests for it.
I feel like this is "90-100% coverage" territory where the complexity of the tests don't yield proportial benefits.

Reply:
I agreed that testing `version` isn't worth the effort required at this point.  It's too bad there isn't a more testable way to do VCS version stamping in Go. :(